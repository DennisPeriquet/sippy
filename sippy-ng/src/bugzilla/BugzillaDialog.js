import { Button, Divider, Tooltip, Typography } from '@material-ui/core'
import { Close, Info } from '@material-ui/icons'
import BugTable from './BugTable'
import bugzillaURL from './BugzillaUtils'
import Dialog from '@material-ui/core/Dialog'
import DialogContent from '@material-ui/core/DialogContent'
import DialogTitle from '@material-ui/core/DialogTitle'
import PropTypes from 'prop-types'
import React, { Fragment } from 'react'

export const LINKED_BUGS = `Linked bugs are bugs that mention the failing test name
and are targeted to the release being reported on.`

export const ASSOCIATED_BUGS = `Associated bugs are bugs that mention the failing test
name but are not targeted to the release being reported on.`

/**
 * BugzillaDialog shows the bugs both linked and associated with a
 * job or test. It also has a link to prefill in a new bug.
 */
export default function BugzillaDialog(props) {
  return (
    <Fragment>
      <Dialog
        scroll="paper"
        style={{ verticalAlign: 'top' }}
        maxWidth="md"
        fullWidth={true}
        open={props.isOpen}
        onClose={props.close}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title" style={{ textAlign: 'right' }}>
          <Button startIcon={<Close />} onClick={props.close} />
        </DialogTitle>
        <DialogContent>
          <Typography variant="h5">{props.item.name}</Typography>
          <Divider />

          <Typography variant="h6" style={{ margin: 20 }}>
            Linked Bugs
            <Tooltip title={LINKED_BUGS}>
              <Info />
            </Tooltip>
          </Typography>

          <BugTable bugs={props.item.bugs} />

          <Typography variant="h6" style={{ margin: 20 }}>
            Associated Bugs
            <Tooltip title={ASSOCIATED_BUGS}>
              <Info />
            </Tooltip>
          </Typography>

          <BugTable bugs={props.item.associated_bugs} />

          <Button
            target="_blank"
            href={bugzillaURL(props.release, props.item)}
            variant="contained"
            color="primary"
            style={{ marginTop: 20 }}
          >
            Open a new bug
          </Button>
        </DialogContent>
      </Dialog>
    </Fragment>
  )
}

BugzillaDialog.defaultProps = {
  item: {
    name: '',
    bugs: [],
    associated_bugs: [],
  },
  classes: {},
}

BugzillaDialog.propTypes = {
  release: PropTypes.string,
  item: PropTypes.object.isRequired,
  classes: PropTypes.object,
  isOpen: PropTypes.bool,
  close: PropTypes.func,
}
