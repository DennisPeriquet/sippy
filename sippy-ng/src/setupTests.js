/** @jest-environment setup-polly-jest/jest-environment-node */

import '@testing-library/jest-dom'
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import fetch from 'node-fetch'
import path from 'path'
import { setupPolly } from 'setup-polly-jest'
import toDiffableHtml from 'diffable-html'

// https://github.com/mui-org/material-ui/issues/21293
export const withoutMuiID = (wrapper) => toDiffableHtml(wrapper.html()
  .replace(/id="mui-[0-9]*"/g, '')
  .replace(/aria-labelledby="(mui-[0-9]* *)*"/g, ''))

Enzyme.configure({ adapter: new Adapter() })

// To stub out fetch(), we need to use node-fetch and this line:
global.fetch = fetch

// See: https://github.com/vuejs/vue-test-utils/issues/974
global.requestAnimationFrame = cb => cb()

// When injecting multiple filters, material table complains.
const originalError = console.error.bind(console.error)

global.console.error = (log) => {
  if (!log.toString().includes('Warning: Failed')) {
    originalError(log.toString())
  }
}

// Set API URL for Sippy
process.env.REACT_APP_API_URL = 'http://localhost:8080'

// Default PollyJS mode is replay, set to record to record
// new API calls.
if (!process.env.POLLY_MODE) {
  process.env.POLLY_MODE = 'replay'
}

export const setupDefaultPolly = () => {
  const context = setupPolly({
    mode: process.env.POLLY_MODE,
    adapters: [require('@pollyjs/adapter-node-http')],
    persister: require('@pollyjs/persister-fs'),
    persisterOptions: {
      fs: {
        recordingsDir: path.resolve(__dirname, '../__recordings__')
      }
    }
  })
  return context
}

export const expectLoadingPage = (wrapper) => expect(wrapper.find('p').contains('Loading...'))

jest.mock('react-chartjs-2', () => ({
  Bar: () => null,
  Chart: () => null,
  ChartComponent: () => null,
  Doughnut: () => null,
  Line: () => null,
  PolarArea: () => null
}))
