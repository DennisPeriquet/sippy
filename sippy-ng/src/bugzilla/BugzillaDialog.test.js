import { act } from '@testing-library/react'
import { mount } from 'enzyme'
import { withoutMuiID } from '../setupTests'
import BugzillaDialog from './BugzillaDialog'
import React from 'react'

const item = {
  name: '[sig-arch][Late] operators should not create watch channels very often [Suite:openshift/conformance/parallel]',
  bugs: [
    {
      id: 1979966,
      status: 'NEW',
      last_change_time: '2021-08-02T18:09:41Z',
      summary:
        'workers-rhel7 job is permanently failing [periodic-ci-openshift-release-master-nightly-4.8-e2e-aws-workers-rhel7]',
      affects_versions: ['4.11', '4.12'],
      fix_versions: ['4.13.0'],
      components: ['Installer'],
      url: 'https://bugzilla.redhat.com/show_bug.cgi?id=1979966',
    },
  ],
}

describe(BugzillaDialog, () => {
  let wrapper

  beforeEach(async () => {
    await act(async () => {
      wrapper = mount(
        <BugzillaDialog
          release="4.8"
          isOpen={true}
          close={() => {}}
          item={item}
        />
      )
    })
    wrapper.update()
  })

  it('shows the name of the test', async () => {
    expect(wrapper.find('h5').text()).toContain(item.name)
  })

  it('shows linked bugs', async () => {
    expect(
      wrapper
        .find('a[href="https://bugzilla.redhat.com/show_bug.cgi?id=1979966"]')
        .exists()
    ).toBeTruthy()
  })

  it('has a link to open a new bug', async () => {
    expect(
      wrapper.find('a[href*="bugzilla.redhat.com"] > .MuiButton-label').text()
    ).toContain('Open a new bug')
  })
})
