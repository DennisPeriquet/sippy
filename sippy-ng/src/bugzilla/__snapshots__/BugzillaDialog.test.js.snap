// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BugzillaDialog renders correctly 1`] = `
<div role="presentation"
     class="MuiDialog-root"
     style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px; vertical-align: top;"
>
  <div class="MuiBackdrop-root"
       aria-hidden="true"
       style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
  </div>
  <div tabindex="0"
       data-test="sentinelStart"
  >
  </div>
  <div class="MuiDialog-container MuiDialog-scrollPaper"
       style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
       role="none presentation"
       tabindex="-1"
  >
    <div class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthMd MuiDialog-paperFullWidth MuiPaper-elevation24 MuiPaper-rounded"
         role="dialog"
         aria-labelledby="form-dialog-title"
    >
      <div class="MuiDialogTitle-root"
           id="form-dialog-title"
           style="text-align: right;"
      >
        <h2 class="MuiTypography-root MuiTypography-h6">
          <button class="MuiButtonBase-root MuiButton-root MuiButton-text"
                  tabindex="0"
                  type="button"
          >
            <span class="MuiButton-label">
              <span class="MuiButton-startIcon MuiButton-iconSizeMedium">
                <svg class="MuiSvgIcon-root"
                     focusable="false"
                     viewbox="0 0 24 24"
                     aria-hidden="true"
                >
                  <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z">
                  </path>
                </svg>
              </span>
            </span>
            <span class="MuiTouchRipple-root">
            </span>
          </button>
        </h2>
      </div>
      <div class="MuiDialogContent-root">
        <h5 class="MuiTypography-root MuiTypography-h5">
          [sig-arch][Late] operators should not create watch channels very often [Suite:openshift/conformance/parallel]
        </h5>
        <hr class="MuiDivider-root">
        <h6 class="MuiTypography-root MuiTypography-h6"
            style="margin: 20px;"
        >
          Linked Bugs
          <svg class="MuiSvgIcon-root"
               focusable="false"
               viewbox="0 0 24 24"
               aria-hidden="true"
               title="Linked bugs are bugs that mention the failing test name
and are targeted or found in the release being reported on."
          >
            <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z">
            </path>
          </svg>
        </h6>
        <div class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
             style="margin-top: 20px;"
        >
          <table class="MuiTable-root"
                 aria-label="bug-table"
          >
            <thead class="MuiTableHead-root">
              <tr class="MuiTableRow-root MuiTableRow-head">
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Bug ID
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Summary
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Status
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Component
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Found In
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Target Release
                </th>
              </tr>
            </thead>
            <tbody class="MuiTableBody-root">
              <tr class="MuiTableRow-root">
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                    scope="row"
                >
                  <a href="https://bugzilla.redhat.com/show_bug.cgi?id=1979966">
                    1979966
                  </a>
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  workers-rhel7 job is permanently failing [periodic-ci-openshift-release-master-nightly-4.8-e2e-aws-workers-rhel7]
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  NEW
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  Installer
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  4.8.0
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <h6 class="MuiTypography-root MuiTypography-h6"
            style="margin: 20px;"
        >
          Associated Bugs
          <svg class="MuiSvgIcon-root"
               focusable="false"
               viewbox="0 0 24 24"
               aria-hidden="true"
               title="Associated bugs are bugs that mention the failing test
name but are not targeted or found in the release being reported on."
          >
            <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z">
            </path>
          </svg>
        </h6>
        <div class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
             style="margin-top: 20px;"
        >
          <table class="MuiTable-root"
                 aria-label="bug-table"
          >
            <thead class="MuiTableHead-root">
              <tr class="MuiTableRow-root MuiTableRow-head">
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Bug ID
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Summary
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Status
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Component
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Found In
                </th>
                <th class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                    scope="col"
                >
                  Target Release
                </th>
              </tr>
            </thead>
            <tbody class="MuiTableBody-root">
              <tr class="MuiTableRow-root">
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                    scope="row"
                >
                  <a href="https://bugzilla.redhat.com/show_bug.cgi?id=1989100">
                    1989100
                  </a>
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  periodic-ci-openshift-release-master-ci-4.8-e2e-openstack-ovn
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  NEW
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  Installer
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                </td>
                <td class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall">
                  ---
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <a class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
           tabindex="0"
           aria-disabled="false"
           target="_blank"
           href="https://bugzilla.redhat.com/enter_bug.cgi?classification=Red%20Hat&amp;product=OpenShift%20Container%20Platform&amp;cf_internal_whiteboard=trt&amp;short_desc=%5Bsig-arch%5D%5BLate%5D%20operators%20should%20not%20create%20watch%20channels%20very%20often%20%5BSuite%3Aopenshift%2Fconformance%2Fparallel%5D&amp;comment=%0A%5Bsig-arch%5D%5BLate%5D%20operators%20should%20not%20create%20watch%20channels%20very%20often%20%5BSuite%3Aopenshift%2Fconformance%2Fparallel%5D%0A%0Ais%20failing%20frequently%20in%20CI%2C%20see%3A%0Ahttps%3A%2F%2Fsippy.ci.openshift.org%2Fsippy-ng%2Ftests%2F4.8%2Fanalysis%3Ftest%3D%255Bsig-arch%255D%255BLate%255D%2520operators%2520should%2520not%2520create%2520watch%2520channels%2520very%2520often%2520%255BSuite%253Aopenshift%252Fconformance%252Fparallel%255D%0A%0AFIXME%3A%20Replace%20this%20paragraph%20with%20a%20particular%20job%20URI%20from%20the%20search%20results%20to%20ground%20discussion.%20%20A%20given%20test%20may%20fail%20for%20several%20reasons%2C%20and%20this%20bug%20should%20be%20scoped%20to%20one%20of%20those%20reasons.%20%20Ideally%20you'd%20pick%20a%20job%20showing%20the%20most-common%20reason%2C%20but%20since%20that's%20hard%20to%20determine%2C%20you%20may%20also%20chose%20to%20pick%20a%20job%20at%20random.%20%20Release-gating%20jobs%20(release-openshift-...)%20should%20be%20preferred%20over%20presubmits%20(pull-ci-...)%20because%20they%20are%20closer%20to%20the%20released%20product%20and%20less%20likely%20to%20have%20in-flight%20code%20changes%20that%20complicate%20analysis.%0A%0AFIXME%3A%20Provide%20a%20snippet%20of%20the%20test%20failure%20or%20error%20from%20the%20job%20log%0A&amp;version=4.9&amp;cc=sippy@dptools.openshift.org"
           style="margin-top: 20px;"
        >
          <span class="MuiButton-label">
            Open a new bug
          </span>
          <span class="MuiTouchRipple-root">
          </span>
        </a>
      </div>
    </div>
  </div>
  <div tabindex="0"
       data-test="sentinelEnd"
  >
  </div>
</div>
`;
